### PR TITLE
Making sure header files are properly defined and terminated

### LINKED ISSUE #1542: Making sure Header Files are Properly Defined and Terminated
Recently, I've realized that my habit of deleting any empty lines at the end of a file could cause problems with many historical Unix utilities. So, I've decided to take it upon myself to go through every header file in chiventure has a newline at the EOF and is properly defined with `#ifndef ... #define ... #endif`

### PR DESCRIPTION
**Pull request by bvvill**
As described in #1542

### CODE REVIEW

File: include/ui/ui_ctx.h [hunk 1]
```diff
@@ -107,6 +107,4 @@ void toggle_map(chiventure_ctx_t *ctx);
 void layout_switch(chiventure_ctx_t *ctx);
```
Comment (sglenn-uchi) [inline include/ui/ui_ctx.h]: very small nitpick would be to remove one of the extra lines for consistency (1 line between last code line and #endif)

File: include/wdl/load_game.h [hunk 2]
```diff
@@ -60,6 +60,4 @@ void game_set_start(char *room_id, game_t *game);
 void debug_print(game_t *game);
```
Comment (sglenn-uchi) [inline include/wdl/load_game.h]: same nitpick;; cut down to one line break in between code and #endif

File: include/cli/util.h [hunk 3]
```diff
@@ -31,4 +34,6 @@ void case_insensitize(char *input);
  * Returns:
  *   - A lowercased version of the string
  */
-char* case_insensitized_string(char *input);
\ No newline at end of file
+char* case_insensitized_string(char *input);
+
+#endif
```
Comment (sglenn-uchi) [inline include/cli/util.h]: You could potentially add comments after each #endif for stylistic consistency (just "#endif /* FILE_NAME */").

File: include/cli/util.h [hunk 4]
```diff
@@ -1,3 +1,6 @@
+#ifndef UTIL_H
+#define UTIL_H
+
 #include <stdbool.h>
 #include <string.h>
 #include <assert.h>
```
Comment (sglenn-uchi) [inline include/cli/util.h:6]: Very tiny nitpick: you could move this to the top to maintain alphabetical order.

**Review by sglenn-uchi — APPROVED** [review]
Overall I think this looks really good! I didn't see any missing #indef, #define, or #endif statements. A really minor finally touch-up would be to organize the included filenames by alphabetical order. Other than this, though, this PR is ready to go!

**Review by lilyehsani — APPROVED** [review]
Good QOL update! Approved.

**Comment by bvvill** [comment]
To summarize, the changes made in this branch were to ensure that the header (AKA `*.h`) files within chiventure are properly defined and terminated with `#ifndef`, `#define`, and `#endif` statements, followed by a newline at the EOF, such as:
```
#ifndef  HEADER_FILE_H   // insert header file name here
#define HEADER_FILE_H
...
...
...
#endif /* HEADER_FILE_H */

```
This was done in order to avoid possible issues with legacy / historical UNIX utilities and to help optimize compiling so that headers aren't defined multiple times

**Comment by lilyehsani** [comment]
Issue Score: Excellent

Comments:
Great work on this (minor) PR! The issue did not include a descriptive summary (see https://uchicago-cs.github.io/cmsc22000/project/issue-pr-grading.html#descriptive-summary for more details and concrete examples)

**Pull request merged**